.github/workflows/main.yml

```diff
@@ -7,7 +7,17 @@ on:
     branches: [main]
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.9"
+      - uses: pre-commit/action@v2.0.3
+
   build-core:
+    needs: [lint]
```

Comment:
I think build (and test) should run even if lint fails.

Reply:
Okay.

I've never had it hang on GHA, but I don't think I've used prettier, either. I don't see why it would be a problem, though.

Reply:
I think run tests on lint failure is a critical quality of life feature. But I don't really look at the gha tests so it doesn't matter as much.